cli/agent_test.go

```diff
@@ -75,9 +75,9 @@ func TestWorkspaceAgent(t *testing.T) {
 		user := coderdtest.CreateFirstUser(t, client)
 		version := coderdtest.CreateTemplateVersion(t, client, user.OrganizationID, &echo.Responses{
 			Parse: echo.ParseComplete,
-			ProvisionApply: []*proto.Provision_Response{{
-				Type: &proto.Provision_Response_Complete{
-					Complete: &proto.Provision_Complete{
+			ProvisionApply: []*proto.Response{{
```

Comment:
It's not really possible to put in a separate PR, since the protocol changes are fundamental, and without renames nothing builds.

I've split it out into different commits to reduce the noise

Reply:
I admit that I focuses mainly on changes in the CLI package like:

* `echo.ProvisionComplete` -> `echo.ApplyComplete`
* `proto.Provision_Response` -> `proto.Response`
* `completeWithAgent()`

but if these are strongly coupled with the protocol, then please ignore the comment.